src/vellum/workflows/nodes/displayable/tool_calling_node/tests/test_node.py

```diff
@@ -140,3 +140,64 @@ def test_deployment_definition_release_tag_defaults_to_latest():
 
     # THEN the release_tag should default to "LATEST"
     assert deployment_config.release_tag == "LATEST"
+
+
+def test_create_tool_router_node_preserves_user_chat_history_block():
+    """
+    Test that create_tool_router_node doesn't duplicate chat_history blocks when user provides one.
+    """
+    from vellum.client.types.variable_prompt_block import VariablePromptBlock
+
+    # GIVEN a user-provided chat history block
+    user_chat_history_block = VariablePromptBlock(
+        block_type="VARIABLE",
+        input_variable="chat_history",
+        state=None,
+        cache_config=None,
+    )
+
+    original_blocks: List[PromptBlock] = [user_chat_history_block]
+
+    # WHEN creating a tool router node with that block
+    create_tool_router_node(
```

Comment:
Don't call this util directly. Instead, define a ToolCallingNode with the VariablePromptBlock in blocks, pass in a chat history input, then assert the blocks we send to the execute prompt API